Code/GraphMol/MolStandardize/Normalize.cpp

```diff
@@ -98,30 +97,31 @@ ROMol *Normalizer::normalize(const ROMol &mol) {
   return outmol;
 }
 
-boost::shared_ptr<ROMol> Normalizer::normalizeFragment(
+ROMOL_SPTR Normalizer::normalizeFragment(
     const ROMol &mol,
-    const std::vector<std::shared_ptr<ChemicalReaction>> &transforms) {
-  boost::shared_ptr<ROMol> nfrag(new ROMol(mol));
+    const std::vector<std::shared_ptr<ChemicalReaction>> &transforms) const {
+  ROMOL_SPTR nfrag(new ROMol(mol));
   MolOps::fastFindRings(
       *nfrag);  // this doesn't do anything if rings are already there
+  std::set<std::string> seenProductSmiles;
   for (unsigned int i = 0; i < MAX_RESTARTS; ++i) {
-    bool loop_brake = false;
+    bool loop_break = false;
     // Iterate through Normalization transforms and apply each in order
     for (auto &transform : transforms) {
-      boost::shared_ptr<ROMol> product =
-          this->applyTransform(nfrag, *transform);
-      if (product != nullptr) {
+      SmilesMolPair product = applyTransform(nfrag, *transform);
+      if (!product.first.empty() and !seenProductSmiles.count(product.first)) {
```

Comment:
A bit of Python leaked into your C++:
```suggestion
      if (!product.first.empty() && !seenProductSmiles.count(product.first)) {
```

Reply:
Ouch, `gcc` doesn't complain about that so I didn't realize. That balances all end-of-line `;` I put in Python, I guess.

Reply:
I can't tell you how many times I've done this.

Reply:
maybe gcc is not totally wrong? https://en.cppreference.com/w/cpp/keyword/and

Reply:
@rvianello You can enable keyword Boolean operators in MSVC through appropriate switches (and I did to get, for example, SireMol to compile under MSVC):
https://stackoverflow.com/questions/24414124/why-does-vs-not-define-the-alternative-tokens-for-logical-operators
but my `and` really leaked in inadvertently - I would not use it in C++ intentionally, especially since those MSVC command line switches do a lot of things apart from dealing with Boolean operators..

Reply:
yes, I also wouldn't use it, but I recalled it was legal and I was surprised MSVC didn't support it. Thanks to the link to SO.